### DISCUSSION TITLE
failed to set TCP sender port ID to specific value !

### DISCUSSION THREAD

**Discussion by mahmoudAboelfath** [dannagle/PacketSender#249] [Q&A]
i want to send a TCP package with specific port ID, i am setting the port id through file/setting/TCP server port 
however when sending the data, the port value is random not the value i set ! 
what am i missing 
please note same way is working fine with UDP where i could set the sender port ID to a specific value, it only fails with TCP.

**Comment by dannagle** [comment]
The TCP client from the GUI is dynamic. It always requests a port from the OS to send. The port you are setting in the Settings area is the built-in server where Packet Sender listens for incoming connections. The UDP port setting is both client and server because that's required for Packet Sender to receive UDP responses. 

If you really need to set the source port for TCP, that can be done on the command line with the `--bind` option.
